lib/JsonpExportMainTemplatePlugin.js

```diff
@@ -4,25 +4,34 @@
 */
 var ConcatSource = require("webpack-sources").ConcatSource;
 
-function JsonpExportMainTemplatePlugin(name) {
-	this.name = name;
+function JsonpExportMainTemplatePlugin(names) {
+	if (names && names.constructor !== Array) {
+		names = [names];
+	}
+	this.names = names.length > 0 ? names : undefined;
 }
 module.exports = JsonpExportMainTemplatePlugin;
 JsonpExportMainTemplatePlugin.prototype.apply = function(compilation) {
 	var mainTemplate = compilation.mainTemplate;
 	compilation.templatesPlugin("render-with-entry", function(source, chunk, hash) {
-		var name = mainTemplate.applyPluginsWaterfall("asset-path", this.name || "", {
-			hash: hash,
-			chunk: chunk
-		});
-		return new ConcatSource(name + "(", source, ");");
+		return new ConcatSource(this.names.map(function(nameSrc) {
+			var name = mainTemplate.applyPluginsWaterfall("asset-path", nameSrc || "", {
+				hash: hash,
+				chunk: chunk
+			});
+			return name + " && " + name + "(", source, ");";
+		}).join(""));
 	}.bind(this));
 	mainTemplate.plugin("global-hash-paths", function(paths) {
-		if(this.name) paths.push(this.name);
+		if(this.names) paths = paths.concat(this.names);
```

Comment:
https://github.com/webpack/webpack/pull/2984/commits/8695ce7bac318afa5f96b2ba68cd74afe657e8d5 fixes the ones in this PR